pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionaryMultiColumnGroupKeyGenerator.java

```diff
@@ -177,6 +305,29 @@ private int getGroupIdForKey(FixedIntArray keyList) {
     return groupId;
   }
 
+  /**
+   * Helper method to get or create a list of group-id for a list of group key.
+   *
+   * @param keysList Group keys, that is a list of list of objects to be grouped
+   * @return Group ids
+   */
+  private int[] getGroupIdsForKey(int[][] keysList) {
+    List<Integer> groupIds = new ArrayList<>();
+    getGroupIdsForKeyHelper(keysList, new int[keysList.length], 0, groupIds);
+    return groupIds.stream().mapToInt(i->i).toArray();
+  }
+
+  private void getGroupIdsForKeyHelper(int[][] keysList, int[] groupKeyIds, int level, List<Integer> groupIds) {
+    if (level == keysList.length) {
+      groupIds.add(getGroupIdForKey(new FixedIntArray(groupKeyIds)));
```

Comment:
(Critical) Need to make a copy of `groupKeyIds`, because it can be inserted into the map